packages/tuono-router/src/components/RouterProvider.tsx

```diff
@@ -1,51 +1,32 @@
-import React from 'react'
-import type { ReactNode, JSX } from 'react'
+import type { JSX } from 'react'
+import { Suspense } from 'react'
 
-import { useListenBrowserUrlUpdates } from '../hooks/useListenBrowserUrlUpdates'
-import { useInitRouterStore } from '../hooks/useRouterStore'
 import type { ServerProps } from '../types'
 import type { Router } from '../router'
 
-import { getRouterContext } from './RouterContext'
+import { RouterContextProvider } from './RouterContext'
 import { Matches } from './Matches'
 
-interface RouterContextProviderProps {
-  router: Router
-  children: ReactNode
-}
-
-function RouterContextProvider({
-  router,
-  children,
-}: RouterContextProviderProps): JSX.Element {
-  // Allow the router to update options on the router instance
-  router.update({ ...router.options } as Parameters<typeof router.update>[0])
-
-  const routerContext = getRouterContext()
-
-  return (
-    <React.Suspense>
-      <routerContext.Provider value={router}>{children}</routerContext.Provider>
-    </React.Suspense>
-  )
-}
-
 interface RouterProviderProps {
   router: Router
   serverProps?: ServerProps
 }
 
+/**
+ * This component is the tuono app entry point
+ */
 export function RouterProvider({
   router,
   serverProps,
 }: RouterProviderProps): JSX.Element {
-  useInitRouterStore(serverProps)
-
-  useListenBrowserUrlUpdates()
-
   return (
-    <RouterContextProvider router={router}>
-      <Matches serverSideProps={serverProps?.props} />
-    </RouterContextProvider>
+    <Suspense>
```

Comment:
IMHO the Suspense should not be added here. 
It's likely that a user wants to customise the `fallback` behaviour,
so I think is better if we handle this directly in `tuono` package

Reply:
This Suspense is required by react. Without it we get a console error.

Reply:
Gotcha, we might end up to add a prop to allow `fallback` customisation in the future.